data/abilities.ts

```diff
@@ -3514,11 +3517,14 @@ export const Abilities: {[abilityid: string]: AbilityData} = {
 		num: 100,
 	},
 	stalwart: {
-		onModifyMove(move) {
-			// this doesn't actually do anything because ModifyMove happens after the tracksTarget check
-			// the actual implementation is in Battle#getTarget
-			move.tracksTarget = true;
+		onModifyMovePriority: 1,
+		onModifyMove(move, pokemon, target) {
+			if (!(move.target === 'scripted')) {
+				move.tracksTarget = true;
+			}
 		},
+		// the actual implementation is in Battle#getTarget
```

Comment:
I'd change this to something like "most of the implementation is done in Battle#getTarget". Move the comment inside of onModifyMove (above the if statement).